tests/unit/parallel/executors/shared_parallel_executor.cpp

```diff
@@ -32,6 +33,17 @@ struct shared_parallel_executor
     {
         return hpx::async(std::forward<F>(f));
     }
+
+    template <typename F>
+    typename std::remove_reference<
+        typename hpx::util::result_of<
+            typename hpx::util::decay<F>::type()
+        >::type
+    >::type
```

Comment:
>> I tried solving it in the trait, but was not able to do it there. If you
>> have an idea how this could be formulated, I'd appreciate any ideas.
>
> I am still trying to wrap my around why this is needed in the first place.

`shared_future<>::get()` returns a `const&` to the result. [This code](https://github.com/STEllAR-GROUP/hpx/blob/master/hpx/parallel/executors/executor_traits.hpp#L158) extracts the result, passing it along, the shared_future itself however goes out of scope.